### PR TITLE
Add RegisterView which allows to create new user profile (issue #10).

### PR DESCRIPTION
**Pull request by mlen108**

### CODE REVIEW

File: profiles/views.py [hunk 1]
```diff
@@ -1,9 +1,12 @@
 from django.conf import settings
 from django.contrib import messages
+from django.contrib.auth import login, authenticate
 from django.contrib.auth.decorators import login_required
 from django.contrib.sites.models import Site
 from django.core.urlresolvers import reverse, get_callable
-from django.views.generic import TemplateView, UpdateView
+from django.views.generic import TemplateView, CreateView, UpdateView
+
+from incuna.utils.unique_id import generate_id
```
Comment (jturnbull) [inline profiles/views.py]: Does this need to come from incuna? It's not open-sourced

File: profiles/urls.py [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
 from django.conf.urls.defaults import *
 
-from profiles.views import ProfileEdit, ProfileView
+from profiles.views import ProfileEdit, ProfileView, RegisterView
 
 
 urlpatterns = patterns('profiles.views',
-    url(r'^$', ProfileView.as_view(), name='profile'),
-    url(r'^edit/$', ProfileEdit.as_view(), name='profile-edit'),
+    url(r'^profile/$', ProfileView.as_view(), name='profile'),
```
Comment (jturnbull) [inline profiles/urls.py]: Do we need to enforce profile/ on these URLs?

File: profiles/urls.py [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
+    url(r'^profile/edit/$', ProfileEdit.as_view(), name='profile-edit'),
 )
 
+if getattr(settings, 'REGISTRATION_ON_VIEW', False):
```
Comment (maxpeterson) [inline profiles/urls.py]: Can we call the setting REGISTRATION_ENABLED

File: profiles/urls.py [hunk 4]
```diff
@@ -1,10 +1,15 @@
+from django.conf import settings
 from django.conf.urls.defaults import *
 
-from profiles.views import ProfileEdit, ProfileView
+from profiles.views import ProfileEdit, ProfileView, RegisterView
 
 
 urlpatterns = patterns('profiles.views',
```
Comment (maxpeterson) [inline profiles/urls.py]: Remove profiles.views from patterns

File: profiles/views.py (hunk 1 above)
Comment (maxpeterson) [inline profiles/views.py]: Copy generate_id to this app.

**Pull request merged**